### PR TITLE
CORE-4: added schemas for groups and subjects

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/common_swagger_api/schema/groups.clj [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+(defn group-with-detail [group-descriptor]
+  (assoc (group group-descriptor)
+    (s/optional-key :detail)
+    (describe (group-detail group-descriptor) (str "Detailed information about the " group-descriptor))))
+
+(defn group-list [group-descriptor plural-group-descriptor]
+  {:groups (describe [(group group-descriptor)] (str "The list of " plural-group-descriptor " in the result set"))})
+
+(defn group-members [group-descriptor]
+  {:members (describe [subjects/Subject] (str "The list of " group-descriptor " members"))})
```
Comment (slr71) [inline src/common_swagger_api/schema/groups.clj:93]: The idea behind these functions is that the service that is using these schemas will call them and pass the result to `schema.core/defschema` to actually define the schema, for example:

```clojure
(s/defschema BaseGroup (group-schema/base-group "group"))
(s/defschema Group (group-schema/group "group"))
(s/defschema GroupUpdate (group-schema/group-update "group"))
(s/defschema GroupStub (group-schema/group-stub "group"))
(s/defschema GroupWithDetail (group-schema/group-with-detail "group"))
(s/defschema GroupList (group-schema/group-list "group" "groups"))
(s/defschema GroupMembers (group-schema/group-members "group"))
(def GroupMembersUpdate group-schema/GroupMembersUpdate)
(def GroupMembersUpdateResponse group-schema/GroupMembersUpdateResponse)
```

File: project.clj [hunk 2]
```diff
@@ -5,8 +5,10 @@
...
                         ["snapshots" :clojars]]
-  :plugins [[test2junit "1.2.2"]]
   :dependencies [[org.clojure/clojure "1.8.0"]
                  [cheshire "5.6.3"]
                  [metosin/compojure-api "1.1.8"]
-                 [org.cyverse/clojure-commons "2.8.1"]])
+                 [org.cyverse/clojure-commons "2.8.1"]]
+  :eastwood {:linters [:wrong-arity :wrong-ns-form :wrong-pre-post :wrong-tag :misplaced-docstrings]}
+  :plugins [[test2junit "1.2.2"]
+            [jonase/eastwood "0.3.4"]])
```
Comment (slr71) [inline project.clj:14]: I added eastwood for my own sanity here. 😸

**Comment by ash3rz** [comment]
(Not an expert but) LGTM, it's almost a complete match to the `Group` and `Subject` models I added in the UI 😄

File: src/common_swagger_api/schema/groups.clj [hunk 3]
```diff
@@ -0,0 +1,112 @@
+(ns common-swagger-api.schema.groups
+  (:use [common-swagger-api.schema :only [describe NonBlankString ->optional-param]])
+  (:require [common-swagger-api.schema.subjects :as subjects]
+            [schema.core :as s]))
+
+(defn base-group [group-descriptor]
```
Comment (psarando) [inline src/common_swagger_api/schema/groups.clj:6]: 👍

File: src/common_swagger_api/schema/groups.clj (hunk 1 above)
Comment (psarando) [reply src/common_swagger_api/schema/groups.clj:93]: Nice!

**Review by psarando — APPROVED** [review]
LGTM! :shipit: 👍

**Comment by slr71** [comment]
Thanks for the reviews!

**Pull request merged**